ckanext/sitemap/view.py

```diff
@@ -54,13 +33,18 @@ def _remove_file(file):
 
     def _create_file(filename, root):
         log.info("Creating new sitemap.xml file: %s", filename)
-        pkgs = (
-            Session.query(Package)
-            .filter(Package.type == "dataset")
-            .filter(Package.private != True)
-            .filter(Package.state == "active")
-            .all()
-        )
+        
+        context = {'model': model, 'session': model.Session, 'user': c.user or c.author}
+        data_dict = {
+            'q': '*:*',
+            'start': 0,
+            'rows': 500  # Adjust as needed
```

Comment:
A filter query should be added to get datasets and maybe showcases. Otherwise packages with type harvest source would appear.